src/cpp/src/lora_adapter.cpp

```diff
@@ -139,18 +134,30 @@ ConstantMap read_safetensors(const std::filesystem::path& filename) {
 
         OPENVINO_ASSERT(
             ov::shape_size(shape) <= tensor.end_offset_bytes - tensor.begin_offset_bytes,
-            "Tensor shape ", ov::shape_size(shape), " for tensor \"", name, "\" from Safetensors file \"", filename, "\" doesn't match the expected tensor size ",
+            "Tensor shape ", ov::shape_size(shape), " for tensor \"", name, "\" from Safetensors file \"", "safetensor", "\" doesn't match the expected tensor size ",
             tensor.end_offset_bytes - tensor.begin_offset_bytes);
 
         auto type = safetensors_to_ov_element_type(tensor.dtype);
         auto constant =
             std::make_shared<v0::Constant>(type, shape, ptr, nullptr);      // wraps existing memory, no ownership
-        constant->get_rt_info()["__safetensors_buffer_holder"] = buffer;    // to automatically deallocate underlying memory buffer when last constant that holds it is destroyed
+        constant->get_rt_info()["__safetensors_buffer_holder"] = safetensor;    // to automatically deallocate underlying memory buffer when last constant that holds it is destroyed
         tensors[name] = constant;
     }
     return tensors;
 }
 
+// Reads a file with a given filename expecting Safetensors file format.
+// The file data is mmaped to tensor.
+ConstantMap read_safetensors(const std::filesystem::path& filename) {
+    auto safetensor = ov::read_tensor_data(filename);
+
+    return safetensors_init(safetensor);
+}
+
+// Reads a safetensor and creates a constantmap from the memory.
+ConstantMap read_safetensors(const ov::Tensor& safetensor) {
```

Comment:
Why do we need this function? It it whole duplication of `safetensors_init(safetensor)`. We can move second one to public section in this case